src/providers/data_provider_fo.c

```diff
@@ -645,6 +672,9 @@ errno_t be_resolve_server_process(struct tevent_req *subreq,
         return ENOENT;
     }
 
+    state->svc->current_resolved = state->srv;
```

Comment:
Did you double check that you don't have to use fo_ref_server here? In general, we don't handle fo_server well and storing the pointer anywhere is quite dangerous and it needs to be done with lots of caution.

Reply:
Not longer needed